lib/phlex/compiler/method_compiler.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+class Phlex::Compiler::MethodCompiler < Prism::Visitor
+	def initialize(compiler)
+		@compiler = compiler
+		@mutations = []
+		@in_scope = true
```

Comment:
This tracks if we may have gone out of scope via a block to an unknown method. Out of scope, we need to use a conditional to check that we’re in scope at runtime. Example

```ruby
10.times do
  if self == component_instance
    # compiled code here
  else
    # original code here
  end
end
```